src/forest.rs

```diff
@@ -244,9 +242,355 @@ impl Forest {
         mark_dirty_recursive(&mut self.nodes, &self.parents, node);
     }
 
-    /// Computes the layout of the `node` and its children
-    pub(crate) fn compute_layout(&mut self, node: NodeId, size: Size<Option<f32>>) {
-        // TODO: It's not clear why this method is distinct
-        self.compute(node, size)
+    /// Returns the number of elements in the forest.
+    #[inline]
+    pub(crate) fn len(&self) -> usize {
+        // This could be either [nodes], [parents] or [children] since they all must always have
+        // the same length, given the data structure properties.
+        self.nodes.len()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{Forest, NodeData};
+    use crate::geometry::Size;
+    use crate::node::{MeasureFunc, NodeId};
+    use crate::style::FlexboxLayout;
+    use crate::sys::ChildrenVec;
+
+    fn assert_forest_size(forest: &Forest, size: usize) {
+        // This should assert the forest consistency, each vector must be have the same length
+        assert_eq!(forest.nodes.len(), size);
+        assert_eq!(forest.children.len(), size);
+        assert_eq!(forest.parents.len(), size);
+    }
+
+    fn node_measure_eq(node: &NodeData, measure_fn: fn(Size<Option<f32>>) -> Size<f32>) -> bool {
+        match node.measure.as_ref().unwrap() {
+            MeasureFunc::Raw(m) => measure_fn(Size::NONE) == m(Size::NONE),
+            _ => false,
```

Comment:
This warning bothers me, but I'm unsure how to treat it since it only happens with allocations disabled. Any suggestions?

Reply:
Use a `cfg` annotation on the `_` branch to skip compiling it when allocations are disabled :)

Reply:
Done! Thanks.